src/StatPlots.jl

```diff
@@ -17,7 +17,9 @@ import Loess
 
 export groupapply
```

Comment:
Well, at the very least the macro can certainly translate:
`@given df groupapply(func, x,....)` to `groupapply(func, df, x,....)`
But potentially it could be done in a smarter way so that it would also add support a general table and not only DataFrames and allow for some data manipulation (i.e. group by sth other than a column, for example `:SepalLength .> 3`).
One thing that worries me is the order of the arguments. As of now there are several ways to call it (example after the arrow):

```julia
groupapplay(func, df, x) ~> groupapplay(:density, iris, :SepalLength)
groupapplay(func, df, x, y) ~> groupapplay(:locreg, iris, :SepalLength, :SepalWidth)
```

And 2 convenience methods for plotting `y` as a fct of `x`:

```julia
groupapplay(y, df, x) == groupapplay(:locreg, df, x, y)
groupapplay(df, x, y) == groupapplay(:locreg, df, x, y)
```
The rational behind the first convenience method is that, when called in the standard way, the first variable represents what's on the `y` axis. Without `df` in the function call, these two methods would be ambiguous and we'd need to decide which one to keep.